### PR TITLE
Fix catapult build publish

### PR DESCRIPTION
**Pull request by dfuentes**
Before, the git hashes weren't being substituted correctly because they were within single quotes.  This takes them out of the quotes.

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -34,4 +34,4 @@ publish:
       - $(git rev-parse --short HEAD)
       #- latest
     when:
-      branch: master
+      branch: fix-catapult-build-publish
```
Comment (mohit) [inline .drone.yml]: remove this commit, before merging

**Comment by mohit** [comment]
lgtm after branch name for build is set to master

**Comment by dfuentes** [comment]
@mohit Can you give this another look, it has changed quite a bit since first review.  I've tested this build and it seems to be working as expected.

One thing you'll notice is that I kind of broke drone's plugin abstraction (by taking in the Repo object that drone knows about).  This is so that I could reliably get both the commit hash and the branch, without having to resort to flakey bash subcommands.

**Comment by mohit** [comment]
This direction is definitely better, even though it comes at the cost of cleanliness of drone code. 

Talking offline with @dfuentes it seems like there is an issue with the current approach where eng might not change the `when` block for `docker` and `catapult` publish at the same time. This will cause inconsistent behavior. Since we already know that people often forget to update `.drone.yml` this is a serious issue.

two options we considered:

1. Change the `docker` publish plugin to always publish. Issues: Longer build times, especially for node projects. And more importantly, we don't know how docker hub is going to react to so many builds.

2. Break the drone plugin interface to switch the `catapult` plugin based on the when block for `docker` publish. This keeps the interface from an eng prospective the same. Issue: untidy code, and we will need to write a `gitbot` script to remove `when` blocks from catapult plugin config. 

We're going with option 2. 

cc: @nikhilpandit

File: plugin/publish/publish.go [hunk 2]
```diff
@@ -70,8 +70,9 @@ func (p *Publish) Write(f *buildfile.Buildfile, r *repo.Repo) {
 		p.Bintray.Write(f)
 	}
 
-	// Catapult
-	if p.Catapult != nil && match(p.Catapult.GetCondition(), r) {
+	// Catapult (conditioned off of docker's conditional)
+	if p.Docker != nil && p.Catapult != nil && match(p.Docker.GetCondition(), r) {
```
Comment (dfuentes) [inline plugin/publish/publish.go]: @mohit fyi: here is the change to trigger off of docker's conditional

**Pull request merged**